### PR TITLE
[WIP] server: replace eventloop with mpsc channel.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: src/server/server.rs [hunk 1]
```diff
@@ -123,36 +120,41 @@ impl<T: RaftStoreRouter, S: StoreAddrResolver> Server<T, S> {
...
-        Ok(())
-    }
+        loop {
+            match self.msg_receiver.recv().unwrap() {
+                Msg::Quit => break,
+                Msg::SendStore { store_id, msg } => self.send_store(store_id, msg),
+                Msg::ResolveResult { store_id, sock_addr, msg } => {
+                    self.on_resolve_result(store_id, sock_addr, msg)
+                }
+                Msg::CloseConn { .. } => {}
```
Comment (siddontang) [inline src/server/server.rs]: do we have CLoseConn now?

File: tests/raftstore/server.rs [hunk 2]
```diff
@@ -114,8 +114,8 @@ impl Simulator for ServerCluster {
         drop(listener);
 
         // TODO: simplify creating raft server later.
-        let mut event_loop = create_event_loop(&cfg).unwrap();
-        let sendch = SendCh::new(event_loop.channel(), "cluster-simulator");
+        let (tx, rx) = mpsc::sync_channel(cfg.notify_capacity);
```
Comment (siddontang) [inline tests/raftstore/server.rs]: any benchmark for the sync channel vs channel or mio channel?

File: src/server/server.rs (hunk 1 above)
Comment (disksing) [inline src/server/server.rs]: gRPC conn doesn't need to close manually, it is kept now only for compatibility.

Comment (siddontang) [inline src/server/server.rs]: can we remove it here?

Comment (disksing) [inline src/server/server.rs]: Now the `SnapManager` still send this msg.

**Pull request closed without merging**